numpy/lib/recfunctions.py

```diff
@@ -877,6 +902,9 @@ def join_by(key, r1, r2, jointype='inner', r1postfix='1', r2postfix='2',
         key = (key,)
 
     # Check the keys
+    if len(set(key)) != len(key):
+        dup = (x for n,x in enumerate(key) if x in key[n+1:]).next()
```

Comment:
While we're at it, a test might be nice - since that would have caught this mistake